File: app/src/main/java/com/hippo/ehviewer/dao/DownloadDirnameDao.java
package com.hippo.ehviewer.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface DownloadDirnameDao {

    @Query("SELECT * FROM DOWNLOAD_DIRNAME WHERE GID = :gid")
    DownloadDirname load(long gid);

    @Update
    void update(DownloadDirname downloadDirname);

    @Insert
    void insert(DownloadDirname downloadDirname);

    @Query("DELETE FROM DOWNLOAD_DIRNAME WHERE GID = :gid")
    void deleteByKey(long gid);

    @Query("DELETE FROM DOWNLOAD_DIRNAME")
    void deleteAll();
}

File: app/src/main/java/com/hippo/ehviewer/dao/DownloadDirnameDao.kt
package com.hippo.ehviewer.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface DownloadDirnameDao {
    @Query("SELECT * FROM DOWNLOAD_DIRNAME WHERE GID = :gid")
    suspend fun load(gid: Long): DownloadDirname?

    @Upsert
    suspend fun upsert(downloadDirname: DownloadDirname)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertOrIgnore(downloadDirnameList: List<DownloadDirname>)

    @Query("DELETE FROM DOWNLOAD_DIRNAME WHERE GID = :gid")
    suspend fun deleteByKey(gid: Long)

    @Query("SELECT * FROM DOWNLOAD_DIRNAME")
    suspend fun list(): List<DownloadDirname>
}
